web/server/codechecker_server/api/report_server.py

```diff
@@ -1613,6 +1613,7 @@ def getComments(self, report_id):
                     sys_comment = comment_kind_from_thrift_type(
                         ttypes.CommentKind.SYSTEM)
                     if comment.kind == sys_comment:
+                        message = message.replace("'", "\\'")
```

Comment:
I don't think this is enough. What if it contains double quotes, or backticks, or other shell-special characters? Maybe `shlex.quote()` works good for you? Not sure, since I don't understand what `shlex.split()` is used for.

Reply:
@jimis Good point, you are right. I changed it to use the `shlex.quote` function and I've also extended the test case with more special characters.